frame/session/src/historical/storage.rs

```diff
@@ -0,0 +1,138 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2019-2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Validator Set Extracting an iterator from an offchain worker stored list containing historical validatorsets.
+//!
+//! This is used in conjunction with [`ProvingTrie`](super::ProvingTrie) and
+//! the offchain indexing API.
+
+use sp_io::offchain_index;
+use sp_std::prelude::*;
+
+use super::super::{Module as SessionModule, SessionIndex};
+use super::Trait;
+
+const PREFIX: &[u8] = b"historical";
+const LAST_PRUNE: &[u8] = b"historical_last_prune";
+const HEAD: &[u8] = b"historical_head";
+
+pub struct ValidatorSet<T: Trait> {
+    validator_set: Vec<(T::ValidatorId, T::FullIdentification)>,
+}
+
+/// Derive the key used to store the list of validators
+fn derive_key<P: AsRef<[u8]>, T: Trait>(prefix: P, session_index: Vec<u8>) -> Vec<u8> {
```

Comment:
Could this accept a `SessionIndex` instead of a `Vec<u8>` and use `session_index.using_encoded`? If not, I recommend `&[u8]` instead of `Vec<u8>`